libs/blocks/merch-cards/merch-cards.js

```diff
@@ -0,0 +1,256 @@
+    ]);
+  }
+
+  const preferences = {};
+
+  if (settingsEl) {
+    const [filter, limit = 24] = settingsEl
+      .textContent.split(',').map((s) => s.trim());
+
+    settingsEl.parentElement.remove();
+
+    if (filtered) {
+      attributes.filtered = filter;
+      preferences[filter] = parsePreferences(el.querySelectorAll('p'));
+    } else {
+      if (filter) {
+        attributes.filter = filter;
+      }
+      [...el.children].forEach((filterElement) => {
+        const filterName = filterElement.firstElementChild.textContent?.trim();
+        const elements = [...filterElement.lastElementChild.querySelectorAll('p')];
+        preferences[filterName] = parsePreferences(elements);
+      });
+    }
+
+    if (limit) {
+      attributes.limit = limit;
+    }
+  }
+
+  const literalsEl = el.lastElementChild?.firstElementChild;
+  // parse literals
+  const literalSlots = [];
+  if (literalsEl && /filter/.test(literalsEl.querySelector('u')?.innerText)) {
+    literalsEl.querySelectorAll('u').forEach((u) => {
+      const text = u.innerText.trim();
+      if (DIGITS_ONLY.test(text)) {
+        u.outerHTML = '<span data-placeholder="resultCount"></span>';
+      } else if (text === 'search') {
+        u.outerHTML = '<span data-placeholder="searchTerm"></span>';
+      } else if (text === 'filter') {
+        u.outerHTML = '<span data-placeholder="filter"></span>';
+      }
+    });
+    let index = 0;
+    while (literalsEl.firstElementChild) {
+      const literalEl = literalsEl.firstElementChild;
+      let slot;
+      if (literalEl.tagName === 'P') {
+        slot = literalEl;
+      } else if (literalEl.tagName === 'HR') {
+        slot = createTag('div');
+        while (literalEl.nextElementSibling && literalEl.nextElementSibling?.tagName !== 'HR') {
+          slot.appendChild(literalEl.nextElementSibling);
+        }
+        literalEl.nextElementSibling?.remove?.();
+      }
+      literalEl.remove();
+      if (slot) {
+        slot.setAttribute('slot', LITERAL_SLOTS[index]);
+        index += 1;
+      }
+      literalSlots.push(slot);
+    }
+  }
+
+  const type = el.classList[1];
+  const merchCards = createTag('merch-cards', attributes);
+  if (literalSlots.length > 0) {
+    merchCards.append(...literalSlots);
+  } else if (!merchCards.filtered) {
+    merchCards.filtered = 'all';
+  }
+  initMerchCards(config, type, attributes.filtered, el, preferences)
+    .then((async (cardsRoot) => {
+      const cards = [...cardsRoot.children];
+      const batchSize = 3;
+      for (let i = 0; i < cards.length; i += batchSize) {
+        const batch = cards.slice(i, i + batchSize);
+        merchCards.append(...batch);
+        if (i === batchSize) {
+          // avoid layout shift due to size of first cards.
+          merchCards.requestUpdate();
+        }
+        await makePause();
+      }
+      merchCards.displayResult = true;
+    }
+    ));
+
+  const appContainer = document.querySelector('.merch.app');
```

Comment:
What is this `.merch.app` container? Is this driven by authoring?

Reply:
Yes,  basically if a merch block finds a `.merch.app` element in the DOM, it will go stick to it instead of being rendered in-place.

sidenav in cc do the same so that we can achieve the desired layout.

For the catalog page, we have a dumb block serving as a merch.app container.